source/kraken/fill_disruption_from_chaos.cpp

```diff
@@ -474,8 +474,13 @@ struct delete_impacts_visitor : public apply_impacts_visitor {
     bool func_on_vj(nt::VehicleJourney& vj) {
         vj.adapted_validity_pattern = vj.validity_pattern;
         ++ nb_vj_reassigned;
-        vj.remove_impact(impact);
-        for (auto i: vj.get_impacts()) {
+        //it don't seem to be possible to capture an attribute of the object
+        const auto& impact = this->impact;
+        vj.impacted_by.erase(std::remove_if(vj.impacted_by.begin(), vj.impacted_by.end(),
+                [&impact](const boost::weak_ptr<nt::new_disruption::Impact>& i) {
+            return i.lock() == impact;
+        }), vj.impacted_by.end());
```

Comment:
you can use `boost::remove_erase_if` http://www.boost.org/doc/libs/1_57_0/libs/range/doc/html/range/reference/algorithms/new/erase.html